src/mlpack/tests/ann_layer_test.cpp

```diff
@@ -4656,6 +4656,50 @@ TEST_CASE("TransposedConvolutionWeightInitializationTest", "[ANNLayerTest]")
       == (outSize * inSize * kernelWidth * kernelHeight) + outSize);
 }
 
+/**
+ * Simple Test for ChannelShuffle layer.
+ */
+TEST_CASE("ChannelShuffleLayerTest", "[ANNLayerTest]")
+{
+  arma::mat input1, output1, outputExpected1, outputBackward1;
+  ChannelShuffle<> module1(2, 2, 6, 2);
+
+  input1 << 1  << 13 << arma::endr
+         << 2  << 14 << arma::endr
+         << 3  << 15 << arma::endr
+         << 4  << 16 << arma::endr
+         << 5  << 17 << arma::endr
+         << 6  << 18 << arma::endr
+         << 7  << 19 << arma::endr
+         << 8  << 20 << arma::endr
+         << 9  << 21 << arma::endr
+         << 10 << 22 << arma::endr
+         << 11 << 23 << arma::endr
+         << 12 << 24 << arma::endr;
+  input1.reshape(24, 1);
+  outputExpected1 << 1  << 17 << arma::endr
+                  << 2  << 18 << arma::endr
+                  << 3  << 19 << arma::endr
+                  << 4  << 20 << arma::endr
+                  << 13 << 9 << arma::endr
+                  << 14 << 10 << arma::endr
+                  << 15 << 11 << arma::endr
+                  << 16 << 12 << arma::endr
+                  << 5  << 21 << arma::endr
+                  << 6  << 22 << arma::endr
+                  << 7  << 23 << arma::endr
+                  << 8  << 24 << arma::endr;
```

Comment:
Do those values come from another framework?

Reply:
I checked these values using Pytorch. I will link the implementation.

Reply:
https://colab.research.google.com/drive/16ZATOft3wQ9u_IIeyNbupV-ouzuYHK9Z?usp=sharing

Reply:
Can you add a comment right above the `outputExpected1` and note that the values where calculated using PyTorch.